build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/GetMappedFilesThatExist.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (C) 2011 Xamarin, Inc. All rights reserved.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.BuildTools.PrepTasks {
+	// We have a list of files, we want to get the
+	// ones that actually exist on disk.
+	public class GetMappedFilesThatExist : Task
+	{
+		[Required]
+		public string[] CheckedFiles { get; set; }
+
+		[Required]
+		public string[] ResultingFiles { get; set; }
```

Comment:
I don't understand the semantics of `GetMappedFilesThatExist.ResultingFiles`. It's `[Required]`, but it's only used to obtain `ResultingFiles.Length`. I don't understand.

It *appears* to be to verify that e.g. `%(ApiFileDefinition.ClassParseXml)` and `%(ApiFileDefinition.ParameterDescription)` have the same number of values, but I don't see how this could possibly be otherwise -- they *must* have the same number of values (though some may be the empty string).

Reply:
Are you serious? When you write program, do you skip checking arguments sanity because "they must be always correct" ?

Reply:
Until this fix gets merged, anyone who works on MSBuild tasks will have to wait for long time to actually MSBuild tasks start building, thanks to blindly added dependencies in the .csproj.

Reply:
When you write unit tests, do you write unit tests to test the unit test library? Or do you just assume that the unit test library works?

When you write a program, do you test that  **memcpy**(3) behaves as documented, or do you just assume that it works as documented? (Which might actually be [necessary](https://www.imperialviolet.org/2016/06/26/nonnull.html), mind you...)

There is a difference between "validating arguments" and "not trusting the executing environment and adding extra logic to validate what must always be true."